aiohttp/client.py

```diff
@@ -1131,6 +1147,18 @@ def _prepare_headers(self, headers: LooseHeaders | None) -> "CIMultiDict[str]":
                     added_names.add(key)
         return result
 
+    def _get_netrc_auth(self, host: str) -> BasicAuth | None:
+        """
+        Get auth from netrc for the given host.
+
+        This method is designed to be called in an executor to avoid
+        blocking I/O in the event loop.
+        """
+        netrc_obj = netrc_from_env()
+        with suppress(LookupError):
+            return basicauth_from_netrc(netrc_obj, host)
+        return None
```

Comment:
Maybe it's just me, but this seems more sensible to me as a regular try:
```suggestion
        try:
            return basicauth_from_netrc(netrc_obj, host)
        except LookupError:
            return None
```